src/Appveyor.TestLogger/AppveyorLoggerQueue.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Microsoft.VisualStudio.TestPlatform.Extensions.Appveyor.TestLogger
+{
+	using System;
+	using System.Collections.Generic;
+	using System.Net.Http;
+	using System.Text;
+	using System.Threading;
+	using System.Threading.Tasks;
+
+	internal class AppveyorLoggerQueue
+	{
+		private static readonly HttpClient client = new HttpClient();
+
+		/// <summary>
+		/// it is localhost with a random port, e.g. http://localhost:9023/
+		/// </summary>
+		private readonly string appveyorApiUrl;
+
+		private readonly AsyncProducerConsumerCollection<string> queue = new AsyncProducerConsumerCollection<string>();
+		private readonly Task consumeTask;
+		private readonly CancellationTokenSource consumeTaskConcellationTokenSource = new CancellationTokenSource();
+
+		public AppveyorLoggerQueue(string appveyorApiUrl)
+		{
+			this.appveyorApiUrl = appveyorApiUrl;
+			this.consumeTask = ConsumeItemsAsync(consumeTaskConcellationTokenSource.Token);
+		}
+
+		public void Enqueue(string json)
+		{
+			queue.Add(json);
+		}
+
+		public void Flush()
+		{
+			consumeTaskConcellationTokenSource.Cancel();
+			queue.Cancel();
+			try
+			{
+				consumeTask.Wait(TimeSpan.FromSeconds(10));
```

Comment:
Not specifically.  The API calls are to a `localhost` and I expected them to be fast.  Just thought there should be something.  Do you have a suggestion for a better length?